futurex_openedx_extensions/helpers/clickhouse_operations.py

```diff
@@ -0,0 +1,218 @@
+"""Clickhouse helper functions."""
+from __future__ import annotations
+
+import os
+from typing import Any
+
+import yaml  # type: ignore
+from clickhouse_connect import get_client as clickhouse_get_client
+from clickhouse_connect.driver.httpclient import HttpClient
+from clickhouse_connect.driver.query import QueryResult
+from django.conf import settings
+from django.core.paginator import EmptyPage
+
+
+class ClickhouseBaseError(Exception):
+    """Clickhouse base error."""
+
+
+class ClickhouseClientNotConfiguredError(ClickhouseBaseError):
+    """Clickhouse client not configured error."""
+
+
+class ClickhouseClientConnectionError(ClickhouseBaseError):
+    """Clickhouse client connection error."""
+
+
+class ClickhouseQueryParamsError(ClickhouseBaseError):
+    """Clickhouse query parameters error."""
+
+
+class ClickhouseDefaultQueriesError(ClickhouseBaseError):
+    """Clickhouse default queries error."""
+
+
+def get_client() -> HttpClient:
+    """
+    Get Clickhouse client.
+
+    :return: Clickhouse client.
+    :rtype: HttpClient
+    """
+    try:
+        username = settings.FX_CLICKHOUSE_USER
+        password = settings.FX_CLICKHOUSE_PASSWORD
+    except AttributeError as exc:
+        raise ClickhouseClientNotConfiguredError(f'Error getting Clickhouse credentials: {exc}') from exc
+
+    try:
+        client = clickhouse_get_client(host='clickhouse', port=8123, username=username, password=password)
+    except Exception as exc:
+        raise ClickhouseClientConnectionError(f'Error getting Clickhouse client: {exc}') from exc
+
+    return client
+
+
+def escape_param(param_name: str, param_value: Any) -> str:
+    """
+    Escape the parameter value.
+
+    :param param_name: The parameter name.
+    :type param_name: str
+    :param param_value: The parameter value.
+    :type param_value: Any
+    :return: The escaped parameter value.
+    :rtype: str
+    """
+    if isinstance(param_value, str):
+        param_value = param_value.replace("'", "''")
+        return f"'{param_value}'"
+
+    if isinstance(param_value, (int, float)):
+        return str(param_value)
+
+    if isinstance(param_value, (list, tuple)):
+        return '[' + ', '.join(escape_param(param_name, list_item) for list_item in param_value) + ']'
+
+    raise ClickhouseQueryParamsError(f'Unsupported parameter type: ({param_name}: {type(param_value)})')
+
+
+def replace_params(query: str, params: dict) -> str:
```

Comment:
Same concern in regards to `escape`:

Bind params seems to be doing the same process here:

https://github.com/ClickHouse/clickhouse-connect/blob/1550e8976fcad232f6d2ca873e91f86db8e3b7d5/clickhouse_connect/driver/query.py#L369C5-L404

Reply:
completely removed, using default client syntax now `WHERE something == %(variable1)s OR ....`